rostime/include/ros/impl/duration.h

```diff
@@ -52,26 +52,30 @@ namespace ros {
   template<class T>
   T& DurationBase<T>::fromSec(double d)
   {
-    sec = (int32_t)floor(d);
+    int64_t sec64 = (int64_t)floor(d);
+    if (sec64 < INT_MIN || sec64 > INT_MAX)
```

Comment:
You might consider using `INT32_MIN` and `INT32_MAX` instead, reference:

http://en.cppreference.com/w/cpp/types/integer

(here and below)